lib/stats/stat.js

```diff
@@ -0,0 +1,106 @@
+import assert from 'assert';
+import base from 'taskcluster-base';
+
+import * as series from './series';
+
+export default class Stat {
+  constructor(config) {
+    assert(config.influx, 'Must supply an influx configuration');
+    assert(config.workerId, 'Worker ID is required');
+    assert(config.workerType, 'Worker type is required');
+    assert(config.workerGroup, 'Worker group is required');
+    assert(config.workerNodeType, 'Worker instance type is required');
+    assert(config.provisionerId, 'Provisioner ID is required');
+
+    this.influx = new base.stats.Influx(config.influx);
+    this.baseWorkerStats = {
+      workerId: config.workerId,
+      workerGroup: config.workerGroup,
+      workerType: config.workerType,
+      instanceType: config.workerNodeType,
+      provisionerId: config.provisionerId,
+      capacity: config.capacity
+
+    };
```

Comment:
@garndt,
there is next to no overhead in calling `MySeries.reporter(drain, tags)`, you could put it behind a `getter` but it would be way overkill...

Look at the code in base, it's just a quick check (up front) and a tiny bit of memory.